frontend/src/metabase/components/ToolbarButton/ToolbarButton.tsx

```diff
@@ -0,0 +1,67 @@
+import type { ButtonHTMLAttributes, Ref, MouseEvent } from "react";
+import { forwardRef } from "react";
+
+import { color } from "metabase/lib/colors";
+import type { ActionIconProps, IconName } from "metabase/ui";
+import { ActionIcon, Icon, Tooltip } from "metabase/ui";
+
+export type ToolbarButtonProps = {
+  icon?: IconName;
+  "aria-label": string;
+  tooltipLabel?: string;
+  visibleOnSmallScreen?: boolean;
+  isActive?: boolean;
+  hasBackground?: boolean;
+} & ActionIconProps &
+  ButtonHTMLAttributes<HTMLButtonElement>;
+
+export const ToolbarButton = forwardRef(function ToolbarButton(
+  {
+    icon = "unknown",
+    "aria-label": ariaLabel,
+    onClick,
+    tooltipLabel,
+    visibleOnSmallScreen = true,
+    isActive = false,
+    hasBackground = true,
+    children,
+    disabled,
+    ...actionIconProps
+  }: ToolbarButtonProps,
+  ref: Ref<HTMLButtonElement>,
+) {
+  const handleButtonClick = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (onClick && !disabled) {
+      onClick?.(e);
+    }
```

Comment:
there was a reason why that was there - I think it had something to do with cypress tests. I'll remove it for now and see what happens

Reply:
The only reason we want some thing disabled to have a pointer event is we want to have a hover state for it. e.g. when hovering over a disabled button and show some tooltip or something.

Reply:
yep! that was it - the cypress test wasn't registering the `hover` or `realHover` on the element, so the tooltip wouldn't pop up. I want to see if there's a better way to do this because using `pointerEvents: all` just to make the tests happy seems really hacky to me

Reply:
I believe we could achieve the same thing by wrapping the disable element with a div or a span or some sort and put the tooltip there instead.

Reply:
ah yeah I think we can use a `Box` here